### DISCUSSION TITLE
Minio saves empty directories if there is versioned files.

### DISCUSSION THREAD

**Discussion by valterartur** [minio/minio#15237] [Q&A] [answered]
Hi. 

My request is regarding adding some changes to minio versioning work. 
So we want to replicate our spark db to another minio server. For this purpose we need to enable versioning regarding to documentation. 
And the problem is that versioning causes appearing an empty directories in spark db because of existing versions in this directories. This cause some problem in spark work. 
I didn't found in documentation how to make not keep empty directories even if there is versioned files.
Is it a bug or there is no another way keep versions without saving these empty directories?

**Comment by ravindk89** [comment, marked as answer]
@valterartur this is effectively how versioning and explicit creation of prefixes interact. With Spark this is hard to avoid.

there are a few routes here - one is to have some post process that goes through and cleans up after your spark jobs. 

The other is to use a more recent version of MinIO and take advantage to our improved behavior around excluding paths from versioning:

```
Enable versioning on bucket "mybucket" while excluding versioning on a few select prefixes and all folders.
     Note: this is useful on buckets used with Spark/Hadoop workloads.
     $ mc version enable myminio/mybucket --excluded-prefixes "app1/*/_temporary/,app2/*/_staging/" --exclude-folders
```

So in a sense we already have implemented what you are asking for, by virtue of excluding prefixes within a bucket where you are doing Spark-like workloads but otherwise want versioning.

**Reply by valterartur** [reply]
@ravindk89 thanks for answer a lot! I got it, can you please advice if replication will work properly in case of adding "excluded-prefixes"?

**Reply by harshavardhana** [reply]
Excluded prefixes from versioning won't be replicated thats all.

**Reply by valterartur** [reply]
@harshavardhana thanks for answer :+1:
